### PR TITLE
Fix JIJ output location, add CoreMod to testproject for testing

### PR DESCRIPTION
**Pull request by shartte**

### CODE REVIEW

File: src/main/java/net/neoforged/moddevgradle/dsl/NeoForgeExtension.java [hunk 1]
```diff
@@ -40,6 +46,23 @@ public NeoForgeExtension(Project project) {
         }));
     }
 
+    /**
+     * Adds the necessary dependencies to develop a Minecraft mod to the given source set.
+     * The plugin automatically adds these dependencies to the main source set.
+     */
+    public void addModdingDependenciesTo(SourceSet sourceSet) {
+        var configurations = project.getConfigurations();
+        var sourceSets = project.getExtensions().getByType(SourceSetContainer.class);
```
Comment (Matyrobbrt) [inline src/main/java/net/neoforged/moddevgradle/dsl/NeoForgeExtension.java:55]: There's an util method for this.

File: src/main/java/net/neoforged/moddevgradle/internal/utils/IdeDetection.java [hunk 2]
```diff
@@ -27,4 +33,47 @@ public static boolean isIntelliJSync() {
...
+    public static File getIntellijProjectDir(Project project) {
+        // Always try the root directory first, since it has the highest chance
+        var intellijProjectDir = getIntellijProjectDir(project.getRootDir());
+        if (intellijProjectDir != null) {
+            return intellijProjectDir;
+        }
+
+        // Try every included build
+        for (var includedBuild : project.getGradle().getIncludedBuilds()) {
+            if (!includedBuild.getProjectDir().equals(project.getRootDir())) {
```
Comment (Matyrobbrt) [inline src/main/java/net/neoforged/moddevgradle/internal/utils/IdeDetection.java:52]: What's the purpose of this check?

File: src/main/java/net/neoforged/moddevgradle/internal/utils/IdeDetection.java [hunk 3]
```diff
@@ -27,4 +33,47 @@ public static boolean isIntelliJSync() {
...
+                if (intellijProjectDir != null) {
+                    return intellijProjectDir;
+                }
+            }
+        }
+
+        return null;
+    }
+
+    private static File getIntellijProjectDir(File gradleProjectDir) {
```
Comment (Matyrobbrt) [inline src/main/java/net/neoforged/moddevgradle/internal/utils/IdeDetection.java:63]: Mark as nullable.

File: src/main/java/net/neoforged/moddevgradle/tasks/JarJar.java [hunk 4]
```diff
@@ -36,7 +36,7 @@ public abstract class JarJar extends DefaultTask {
     @Inject
     public JarJar(FileSystemOperations fileSystemOperations) {
         this.fileSystemOperations = fileSystemOperations;
-        this.getOutputDirectory().convention(getProject().getLayout().getBuildDirectory().dir("jarJar/"+getName()));
+        this.getOutputDirectory().convention(getProject().getLayout().getBuildDirectory().dir("generated/jarjar/" + getName()));
```
Comment (Matyrobbrt) [inline src/main/java/net/neoforged/moddevgradle/tasks/JarJar.java:39]: I'm not so sure about `generated/jarJar/taskName`. Can it not just be `generated/taskName`? Or `build/taskName`.

File: testproject/jijtest/src/test/java/jijtest/CoreModTest.java [hunk 5]
```diff
@@ -0,0 +1,14 @@
...
+
+import net.minecraft.world.item.ItemStack;
+import org.junit.jupiter.api.Test;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+public class CoreModTest {
+    @Test
+    void testPresenceOfCoreMod() throws Exception {
+        var field = ItemStack.class.getField("CORE_MOD_MARKER");
```
Comment (Matyrobbrt) [inline testproject/jijtest/src/test/java/jijtest/CoreModTest.java:11]: I'd rather check that `getField` doesn't throw.

**Review by Matyrobbrt — CHANGES_REQUESTED** [review]

File: src/main/java/net/neoforged/moddevgradle/internal/ModDevPlugin.java [hunk 6]
```diff
@@ -651,28 +661,48 @@ private void setupTesting(Project project,
 
         project.afterEvaluate(p -> {
             // Test tasks don't have a provider-based property for working directory, so we need to afterEvaluate it.
-            testTask.configure(task -> task.setWorkingDir(unitTest.getGameDirectory()));
+            testTask.configure(task -> task.setWorkingDir(gameDirectory));
```
Comment (Technici4n) [inline src/main/java/net/neoforged/moddevgradle/internal/ModDevPlugin.java:664]: I think that this part can be removed from `afterEvaluate`.

File: src/main/java/net/neoforged/moddevgradle/internal/ModDevPlugin.java [hunk 7]
```diff
@@ -651,28 +661,48 @@ private void setupTesting(Project project,
...
+
+            // Write out a separate file that has IDE specific VM args, which include the definition of the output directories.
```
Comment (Technici4n) [inline src/main/java/net/neoforged/moddevgradle/internal/ModDevPlugin.java:666]: Please move that to a task.

**Review by Technici4n — CHANGES_REQUESTED** [review]

**Pull request merged**